core/vdbe/execute.rs

```diff
@@ -4517,15 +4525,97 @@ pub fn op_new_rowid(
     else {
         unreachable!("unexpected Insn {:?}", insn)
     };
-    let rowid = {
-        let mut cursor = state.get_cursor(*cursor);
-        let cursor = cursor.as_btree_mut();
-        // TODO: make io handle rng
-        return_if_io!(get_new_rowid(cursor, thread_rng()))
-    };
-    state.registers[*rowid_reg] = Register::Value(Value::Integer(rowid));
-    state.pc += 1;
-    Ok(InsnFunctionStepResult::Step)
+
+    const MAX_ROWID: i64 = i64::MAX;
+    const MAX_ATTEMPTS: u32 = 100;
+
+    loop {
+        match &state.op_new_rowid_state {
+            OpNewRowidState::Start => {
+                state.op_new_rowid_state = OpNewRowidState::CheckingSequential;
+            }
+
+            OpNewRowidState::CheckingSequential => {
+                let current_max = {
+                    let mut cursor = state.get_cursor(*cursor);
+                    let cursor = cursor.as_btree_mut();
+
+                    // Move to last record
+                    return_if_io!(cursor.seek_to_last());
+
+                    return_if_io!(cursor.rowid())
```

Comment:
I think for very big records, this can cause `CursorResult::IO` and an infinite loop as it will start seeking again

Reply:
Maybe just add a FIXME/TODO here